### DISCUSSION TITLE
How do I POST to the AtomPub API?

### DISCUSSION THREAD

**Discussion by ru88s** [freshframework/fresh#766] [Q&A] [answered]
I would like sample code for POSTing to the AtomPub API.
I would like to do a blog post.

```js
import { Handlers } from "$fresh/server.ts";

export default function Home() {
  return (
      <form method="post">
        <button>
          POST
        </button>
      </form>
  );
}

export const handler: Handlers = {
  async POST(req, ctx) {
    console.log("POST");

    return new Response("", {
      status: 303,
      headers: {
        Location: "/",
      },
    });
  },
};
```

**Comment by irvanmalik48** [comment]
Here's how I handle POST data.

```tsx
// some code up here
  async POST(req, ctx) {
    const db = deta.Base("kv");
    const body = (await req.formData()).get("data") as unknown as DetaBaseValue;
    const key = Math.random().toString(36).substring(2, 12);
    const _item = await db.put({
      ...body,
    }, key, {
      expireIn: 604800,
    });

    return ctx.render(
      new Response(JSON.stringify({
        key: key,
        success: true,
      }), {
        headers: {
          "content-type": "application/json",
          "status": "200",
        },
      }),
    );
  },
// some code down here
```
Still has some stuffs to fix tho, and it's still spewing errors but you do `req.formData()` to get the values from the form submission on POST.

**Reply by ru88s** [reply]
Thank you very much.
I would like to post to a blog service that uses the AtomPub API, such as WordPress.
I don't need the code to get the form data, so if you could give me the simplest sample code, that would be great.

**Reply by irvanmalik48** [reply]
You can use `fetch`. And also, doesn't Wordpress have that REST API feature?

**Reply by ru88s** [reply]
It worked!
Thank you very much.

**Comment by lucacasonato** [comment, marked as answer]
To make outbound HTTP requests, use [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/fetch).

**Reply by ru88s** [reply]
I was able to do it.
Thanks for the link too.
